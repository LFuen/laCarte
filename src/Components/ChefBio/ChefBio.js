import React, {Component} from 'react'
import LaCarteContext from '../../context/LaCarteContext'
import './ChefBio.css'


class ChefBio extends Component {

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = LaCarteContext

    render() {

        const {chefs} = this.context
        const chefParam = this.props.match.params.chef
        const individual = chefs.length > 0 ? chefs.find(chef => chef.chef_name === chefParam) : ''

    if(chefs.length > 0) {
        let cuisines = individual.cuisine.substring(1)
        cuisines = cuisines.slice(0, -1)
        let cuisine = cuisines.split(',')

        let schedules = individual.schedule.substring(1)
        schedules = schedules.slice(0, -1)
        let schedule = schedules.split(',')

        if(individual) {
        return (
            <div className='wholeBio'>
                <h1>Chef {individual.chef_name}</h1>
                <div className='chefBio'>
                    <p className='bio'>{individual.bio}</p>
                </div>                
                
                <div className='foodSched'>
                    <div className='cuisine'>
                        <h3>Cuisines</h3>
                        {cuisine.map((origin, idx) => { return (
                        <p key={idx}>{origin}</p>                    
                        )})}                    
                    </div>

                    <div className='schedule' >
                        <h3>Schedule</h3>
                        {schedule.map((days, idx) => { return (
                        <p key={idx}>{days}</p>                    
                        )})}                    
                    </div>                    
                </div>

            </div>
        )            
        } else {
            return (<h3>Sorry, that chef doesn't work here!</h3>)
        }
    } else {
        return (<h3> Loading... </h3>)
    }

    }
}



export default ChefBio
