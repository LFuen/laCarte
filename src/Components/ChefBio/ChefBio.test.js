import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import {BrowserRouter} from 'react-router-dom'
import LaCarteContext from '../../context/LaCarteContext'
import ChefBio from './ChefBio'



describe(`ChefBio Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const bio = renderer.create(<BrowserRouter><ChefBio/></BrowserRouter>)
                expect(bio).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <ChefBio />
                    </LaCarteContext.Provider>                
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})