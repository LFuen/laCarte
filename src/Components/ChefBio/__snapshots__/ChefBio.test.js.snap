// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChefBio Component Snapshot renders the UI as expected 1`] = `
<h3>
   Loading... 
</h3>
`;
